editor/src/components/editor/store/dispatch.tsx

```diff
@@ -747,12 +748,32 @@ function notifyTsWorker(
   }
 }
 
-function removeNonExistingViewReferencesFromState(editorState: EditorState): EditorState {
+function removeNonExistingViewReferencesFromState(
+  oldEditorState: EditorState,
+  editorState: EditorState,
+): EditorState {
+  const oldRootComponents = oldEditorState.jsxMetadataKILLME
```

Comment:
I think at this point we should probably just rename `jsxMetadataKILLME` to `jsxMetadata` :smile: